### PR TITLE
Introduce AlgebroidFromDataTables category constructor

### PR DESCRIPTION
**Pull request by kamalsaleh**

### CODE REVIEW

File: Algebroids/makedoc.g [hunk 1]
```diff
@@ -21,12 +21,16 @@ AutoDoc( rec(
...
+                \usepackage{tikz-cd}
                 \DeclareUnicodeCharacter{2205}{\ensuremath{\emptyset}}
                 \DeclareUnicodeCharacter{2C76}{\ensuremath{\vdash}\!\!}
                 \DeclareUnicodeCharacter{2192}{\ensuremath{\!\!\rightarrow\!}}
                 \DeclareUnicodeCharacter{21AA}{\ensuremath{\!\!\hookrightarrow\!}}
                 \DeclareUnicodeCharacter{21A0}{\ensuremath{\!\!\twoheadrightarrow\!}}
                 \DeclareUnicodeCharacter{2B47}{\ensuremath{\!\!\xrightarrow{\sim}\!}}
+                \DeclareUnicodeCharacter{227B}{\ensuremath{\succ}}
+                \DeclareUnicodeCharacter{2022}{\ensuremath{\bullet}}
+                \DeclareUnicodeCharacter{2297}{\ensuremath{\otimes}}
```
Comment (mohamed-barakat) [inline Algebroids/makedoc.g:33]: This should be produced by PackageJanitor.

File: PreSheaves/gap/PreSheaves.gi [hunk 2]
```diff
@@ -1058,6 +1050,14 @@ InstallMethod( YonedaEmbeddingOfSourceCategory,
     
 end );
 
+##
+InstallOtherMethod( \.,
+        "for a presheaf and a positive integer",
+        [ IsPreSheafCategory, IsPosInt ],
+  
+  { PSh, string_as_int } -> ApplyFunctor( YonedaEmbeddingOfSourceCategory( PSh ), Source( PSh ).( NameRNam( string_as_int ) ) )
```
Comment (mohamed-barakat) [inline PreSheaves/gap/PreSheaves.gi:1058]: Why is this code not covered?

File: Algebroids/gap/AlgebroidFromDataTables.gi [hunk 3]
```diff
@@ -0,0 +1,1919 @@
...
+      
+      o -> CreateObject( B, VertexIndex( UnderlyingVertex( o ) ) )
+    );
+    
+    AddMorphismFunction( eta,
+      
+      { s, alpha, r } -> CreateMorphism( B, s, CoefficientsOfMorphism( A, alpha ), r )
+    );
+    
+    return eta;
```
Comment (mohamed-barakat) [inline Algebroids/gap/AlgebroidFromDataTables.gi:80]: Why is this line not covered?

File: Algebroids/gap/AlgebroidFromDataTables.gi [hunk 4]
```diff
@@ -0,0 +1,1919 @@
...
+              
+          end );
+          
+          AddIsCongruentForMorphisms( PSh,
+            function ( PSh, eta1, eta2 )
+              local B, C;
+              
+              B := Source( PSh );
+              C := Range( PSh );
+
```
Comment (mohamed-barakat) [inline Algebroids/gap/AlgebroidFromDataTables.gi:1297]: No code coverage here as well.

**Review by mohamed-barakat — CHANGES_REQUESTED** [review]
Many Add-methods are not covered. Is it difficult to add tests covering them?

File: Algebroids/makedoc.g (hunk 1 above)
Comment (kamalsaleh) [reply Algebroids/makedoc.g:33]: I will submit a PR to `PackageJanitor`  including the extra latex-declarations after merging this PR.

File: PreSheaves/gap/PreSheaves.gi (hunk 2 above)
Comment (kamalsaleh) [reply PreSheaves/gap/PreSheaves.gi:1058]: Actually, no serious computation in the presheaves category is needed to decide the congruence of morphisms in the quotient category of an algebroid. However, I have updated the PR to include some tests to presheaves.

File: Algebroids/gap/AlgebroidFromDataTables.gi [hunk 5]
```diff
@@ -0,0 +1,1919 @@
...
+    );
+    
+    AddMorphismFunction( eta,
+      
+      { s, alpha, r } -> SumOfMorphisms( A, s, ListN( MorphismDatum( alpha ), BasisOfExternalHom( s, r ), \* ), r )
+    );
+    
+    return eta;
+    
+end );
```
[the remark this answers is not in the document]
Comment (kamalsaleh) [reply Algebroids/gap/AlgebroidFromDataTables.gi:107]: Should be covered now

File: Algebroids/makedoc.g (hunk 1 above)
Comment (mohamed-barakat) [reply Algebroids/makedoc.g:33]: Thanks, it is only important that these lines are created by `PackageJanitor` :)

File: Algebroids/gap/AlgebroidFromDataTables.gi (hunk 3 above)
Comment (kamalsaleh) [reply Algebroids/gap/AlgebroidFromDataTables.gi:80]: should be covered now.

**Comment by mohamed-barakat** [comment]
Thank you very much!

**Pull request merged**